src/nl/hannahsten/texifyidea/run/latex/ui/LatexSettingsEditor.kt

```diff
@@ -297,6 +307,28 @@ class LatexSettingsEditor(private var project: Project?) : SettingsEditor<LatexR
         environmentVariables = EnvironmentVariablesComponent()
         panel.add(environmentVariables)
 
+        expandMacrosEnvVariables = JBCheckBox("Expand macros in environment variables")
+        expandMacrosEnvVariables!!.isSelected = false
+
+        val environmentVariableTextField = environmentVariables.component.textField as ExtendableTextField
+        var envVariableTextFieldMacroSupportExtension: ExtendableTextComponent.Extension? = null
+
+        expandMacrosEnvVariables!!.addItemListener {
+            if (it.stateChange == 1) { // checkbox checked
+                if (envVariableTextFieldMacroSupportExtension != null) {
+                    environmentVariableTextField.addExtension(envVariableTextFieldMacroSupportExtension!!)
+                }
+                else {
+                    MacrosDialog.addMacroSupport(environmentVariableTextField, MacrosDialog.Filters.ALL) { false }
+                    envVariableTextFieldMacroSupportExtension = environmentVariableTextField.extensions.last()
```

Comment:
It is a bit unfortunate that we have to keep a reference to the macro extension like this, but I agree it is probably the most robust way to make sure we remove the right one